test/Autofac.Specification.Test/ContainerBuilderTests.cs

```diff
@@ -38,6 +38,121 @@ public void BuildCallbacksInvokedWhenRegisteredInModuleLoad()
             Assert.Equal(2, module.Called);
         }
 
+        [Fact]
+        public void BuildCallbacksInvokedWhenRegisteredInBeginLifetimeScope()
+        {
+            var called = 0;
+
+            void BuildCallback(ILifetimeScope c)
+            {
+                Assert.False(c is IContainer);
```

Comment:
Boo on xunit for not having `Assert.IsNotAssignableFrom<T>`

Reply:
Agreed, boo on xunit. Not sure if there's a better way to do it that raises a more useful xunit error...